### PR TITLE
BUG: check for nargs <= NPY_MAXARGS in ufunc constructor

### LINKED ISSUE #5672: Segmentation fault when creating a nargs>32 ufunc using frompyfunc in numpy 1.9.2
Title says most of it, here's a minimal example:

faulter.py reads
```python
import numpy as np
print('numpy version:', np.version.version)

def passer(*args):
    pass

for i in (31,32):
    print('\ni =',i)
    vctrzd = np.frompyfunc(passer, i, 1)
    print('about to do it')
    vctrzd(*(np.random.rand(1),)*i)
    print('done')
```
In 1.8.2 this is handled nicely, as when I run this with my /usr/bin/python3 I just get a ValueError:
```
numpy version: 1.8.2

i = 31
about to do it
done

i = 32
about to do it
Traceback (most recent call last):
  File "faulter.py", line 11, in <module>
    vctrzd(*(np.random.rand(1),)*i)
ValueError: Cannot construct an iterator with more than 32 operands (33 were requested)
```

However, I have a virtualenv with the numpy 1.9.2. and there this happens:
```
numpy version: 1.9.2

i = 31
about to do it
done

i = 32
about to do it
Segmentation fault (core dumped)
```

If I run this in an interactive gdb session using
```
gdb env/bin/python3
(gdb) r faulter.py
```
I get
```
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
[New Thread 0x7ffff3b90700 (LWP 9139)]
[New Thread 0x7ffff338f700 (LWP 9140)]
[New Thread 0x7ffff0b8e700 (LWP 9141)]
numpy version: 1.9.2

i = 31
about to do it
done

i = 32
about to do it

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff63e3189 in PyUFunc_GenericFunction (ufunc=0x0, args=0x12, args@entry=0x7fffeb2f6c28, 
    kwds=kwds@entry=0x0, op=0x7fffffffd5d0) at numpy/core/src/umath/ufunc_object.c:2535
2535	numpy/core/src/umath/ufunc_object.c: No such file or directory.
```

I'm running Ubuntu 14.10 and python 3.4.2. The system-wide installation that has numpy 1.8.2. came from the Ubuntu binary repositories where as the virtualenv one with numpy 1.9.2. was installed with pip.

Let me know if more details are needed/helpful.

### PR DESCRIPTION
**Pull request by jaimefrio**
Fixes #5672

### CODE REVIEW

File: numpy/core/src/umath/umathmodule.c [hunk 1]
```diff
@@ -103,6 +103,14 @@ ufunc_frompyfunc(PyObject *NPY_UNUSED(dummy), PyObject *args, PyObject *NPY_UNUS
         PyErr_SetString(PyExc_TypeError, "function must be callable");
         return NULL;
     }
+    if (nin + nout > NPY_MAXARGS) {
+        PyErr_Format(PyExc_ValueError,
```
Comment (charris) [inline numpy/core/src/umath/umathmodule.c]: See above.

**Comment by charris** [comment]
LGTM modulo error message bike shedding.

**Comment by jaimefrio** [comment]
Just wait for the new paint on the bike shed to dry, and it should be ready to merge. ;-)

**Comment by charris** [comment]
Love the color ;) Thanks Jaime.

**Pull request merged**